### ISSUE TITLE
Proposal: Unify embed_dim to 256 via Channel Projection Layer

### ISSUE THREAD

**Issue by riita10069** [autowarefoundation/auto_e2e#23]
## Summary

The current architecture concatenates all 4 backbone stages directly (96+192+384+768 = 1440 channels for Swin) and passes this to View Fusion and downstream modules. This makes the embed_dim backbone-dependent and significantly larger than the industry standard of 256.

This issue proposes adding a Conv1x1 projection layer after multi-scale concatenation to map to a fixed embed_dim (e.g., 256), as discussed in #11: https://github.com/autowarefoundation/auto_e2e/issues/11#issuecomment-4631800289

## Motivation

All major models in this space use embed_dim=256 as their unified feature dimension:

| Model | embed_dim | Reference |
|-------|-----------|-----------|
| BEVFormer | 256 | Li et al., ECCV 2022 |
| UniAD | 256 | Hu et al., CVPR 2023 |
| DETR / Deformable DETR | 256 | Carion et al., ECCV 2020 / Zhu et al., ICLR 2021 |
| PETR / PETRv2 | 256 | Liu et al., ECCV 2022 / ICCV 2023 |
| VAD | 256 | Jiang et al., ICCV 2023 |

Our current 1440-dim has the following implications:

- CrossAttentionViewFusion's multi-head attention scales with embed_dim², making 1440 roughly 31× more expensive than 256
- ConcatViewFusion's Conv1x1(num_views × 1440, 1440) has ~16.6M parameters; with 256 it would be ~0.5M
- Adding a new backbone (e.g., ResNet-50 with 3840 concat channels) requires changing all downstream modules

## Proposed Change

Add a single Conv1x1 in FeatureFusion, after the multi-scale pool + concatenation and before View Fusion:

```python
# In FeatureFusion.__init__:
backbone_channels = 1440  # or 3840 for ResNet-50, etc.
self.channel_proj = nn.Sequential(
    nn.Conv2d(backbone_channels, embed_dim, kernel_size=1),
    nn.GELU()
)

# In FeatureFusion.forward:
fused_per_view = torch.cat((f0, f1, f2, f3), dim=1)  # [B*V, backbone_channels, 7, 7]
fused_per_view = self.channel_proj(fused_per_view)     # [B*V, 256, 7, 7]
```

All downstream modules (ViewFusion, DrivingPolicy, FutureState) would then operate on a fixed 256-dim regardless of backbone choice.

## Impact

- [ ] FeatureFusion: add channel_proj layer, parameterize backbone_channels
- [ ] ViewFusion modules: change embed_dim default from 1440 to 256
- [ ] DrivingPolicy: adjust input dimensions
- [ ] FutureState: adjust channel counts
- [ ] Tests: update expected shapes
- [ ] Design doc: update parameter counts and embed_dim references

## Notes

This is a fairly wide-reaching change since embed_dim=1440 is currently hardcoded throughout. Happy to take this on as a PR if the team agrees on the target dimension. 256 matches the literature, but 512 could also work if we want more capacity. Would appreciate guidance on preference.

**Comment by riita10069** [comment]
A related point: the input resolution and feature map shapes are currently inconsistent across the codebase.

- The benchmark script and backbone registry use `swinv2_tiny_window8_256` (256×256 input, 8×8 final feature maps)
- The NVIDIA dataset parser README documents `(8, 3, 224, 224)` as the visual_tiles shape
- The mock backbone in `conftest.py` (from #28) produces feature maps matching the 256×256 path

If we unify embed_dim as proposed here, it would be a good opportunity to also standardize the input resolution across tests, data parsers, and benchmarks — either 224 or 256, consistently. The backbone's `window_size=8` works cleanly with 256 (256/32=8), while 224 produces fractional feature map sizes at deeper stages.

This isn't a blocker for the embed_dim proposal itself, but worth keeping in mind during implementation so we don't introduce further inconsistency.

**Comment by m-zain-khawaja** [comment]
@riita10069 - I have updated the model to account for multiple backbone feature sizes and have also added ResNet50 to the backbone registry. I ran the inference/run_forward_pass.py script with ResNet50 and it seems to work well. I also updated the Nvidia parser README for 256x256.

To do:

- Update tests
- Update parameter counts and embed_dim reference in design docs

**Comment by m-zain-khawaja** [comment]
Fixes for tests have been merged - thanks @riita10069
